Automattic/woocommerce-payments
# Added Cancel Confirmation Modal

Fixes #5603 

#### Changes proposed in this Pull Request
- Added a confirmation modal when the user chooses to change the status of Order to Cancelled.

#### Testing instructions
1. Place an order.
2. On the admin dashboard 'Edit Order' screen, select "Cancelled" from the Order Status dropdown
3. A confirmation modal should be displayed with the option to "Do Nothing" or "Cancel Order", which warns the merchant that Cancel order will not refund the customer.
4. On click of Do Nothing, the order status dropdown is reset back to its previous status, on click of Cancel Order, it is retained as Cancelled.

Additional Testing Instructions: 
- If the order is not successful, the confirmation modal should not be displayed. e.g. an unpaid order is allowed to be cancelled without showing any confirmation modal.
- Test that there are no regressions in the Refund Confirmation Modal introduced via [PR#4987](https://github.com/Automattic/woocommerce-payments/pull/4987) as that code is also refactored.

-------------------


**Post merge**



- [x] Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) following [these instructions](https://github.com/Automattic/woocommerce-payments/wiki/How-to-write-good-manual-testing-scenarios) : https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions-for-WC-Payments-5.7.0#test-cancel-confirmation-modal
- [x] Not applicable, added E2E tests - Add or update [critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Critical-flows) and [testing instructions for critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows), if applicable.
